### PR TITLE
Overview cards

### PR DESCRIPTION
**Pull request by MartinPirate**

### CODE REVIEW

**Comment by oliveiraigorm** [comment]
I've tested it and I couldn't get all 4 cards in a single row. Also the cards are not aligned with the table 
image

File: src/components/CardOverview.vue [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+									d="M0.988891 4.93102L5.22483 0.714441C5.4429 0.497367 5.79646 0.497367 6.01453 0.714441L10.2505 4.93102C10.6022 5.28119 10.3531 5.87992 9.85561 5.87992H1.38374C0.886258 5.87992 0.637116 5.28119 0.988891 4.93102Z"
+									fill="#47B881"
+								/>
+							</svg>
+
+							<!--							<svg width="11" height="6" viewBox="0 0 11 6" fill="none" xmlns="http://www.w3.org/2000/svg">
+													<path
+														d="M5.66018 5.38911C5.44211 5.60619 5.08855 5.60619 4.87048 5.38911L0.634546 1.17253C0.282771 0.822364 0.531913 0.223632 1.0294 0.223632L9.50127 0.223633C9.99875 0.223633 10.2479 0.822365 9.89612 1.17253L5.66018 5.38911Z"
+														fill="#F64C4C" />
+												</svg>-->
```
Comment (frozenbanana) [inline src/components/CardOverview.vue:30]: Remove if not using

File: src/components/ClientModal.vue [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		toggleModal() {
+			this.$emit("toggle-modal", false);
+		},
+	},
+};
+</script>
+<style>
+.modal-container {
+	width: 80vw !important;
+}
```
Comment (frozenbanana) [inline src/components/ClientModal.vue:83]: avoid using `!important`.

If you must use it please motive why.

File: src/components/ClientModal.vue [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+.ml-22 {
+	margin-left: 22px;
+}
+
+.w-60 {
+	width: 60%;
+}
+
```
Comment (frozenbanana) [inline src/components/ClientModal.vue:135]: I am noting a lot of these utility classes lately.

I think it makes sense to create our own importable CSS file with all of these util classes. God forbid we might use tailwind css. It is ok for now but we should talk about this next week.

**Review by frozenbanana — CHANGES_REQUESTED** [review]
I think we should prioritize having a row of 4 cards. I would increase the width of the table slightly and decrease the card width slightly to make that happen.

Small comment regarding `!important` otherwise good.

File: src/components/ClientModal.vue (hunk 3 above)
Comment (MartinPirate) [reply src/components/ClientModal.vue:135]: I will make the necessary updates in a few

**Comment by oliveiraigorm** [comment]
@MartinPirate I believe you based your branch on the clientView branch. That can create some confusion and conflicts so in the future it's better to base it on the main branch.

**Comment by MartinPirate** [comment]
@oliveiraigorm  yeah I did base my work on the clientView branch, the main page didn't have the table view  which i needed for  further tests, will watch out next time

File: src/components/ClientModal.vue (hunk 2 above)
Comment (MartinPirate) [reply src/components/ClientModal.vue:83]: this was not me  @frozenbanana

File: src/components/ClientModal.vue (hunk 3 above)
Comment (MartinPirate) [reply src/components/ClientModal.vue:135]: I based my changes to @oliveiraigorm  branch which was my bad, wanted to use the table view  for determining the total cards length

File: src/components/CardOverview.vue (hunk 1 above)
Comment (MartinPirate) [reply src/components/CardOverview.vue:30]: noted

File: src/components/CardOverview.vue [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+<style lang="css" scoped>
+.card {
+	box-shadow: 0px 0px 10.6566px rgba(0, 0, 0, 0.15);
+	border-radius: 17.0505px;
+	background: #ffffff;
+	width: 200px;
+	height: 177.96px;
+	flex: 1;
+	/*	margin: 16px;*/
```
Comment (frozenbanana) [inline src/components/CardOverview.vue:101]: Remove if not using

File: src/components/CardOverview.vue [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	</main>
+</template>
+
+<script>
+export default {
+	name: "CardOverview",
+	props: {
+		showData: {
+			type: Boolean,
+		},
```
Comment (frozenbanana) [inline src/components/CardOverview.vue:64]: Not sure when we would like to have `showData`  to be false. 

Consider removing it.

Comment (MartinPirate) [reply src/components/CardOverview.vue:64]: i use it to set the forth card to empty 
will be removed once they decide what will be added on the  card

File: src/components/CardOverview.vue (hunk 1 above)
Comment (mwalbeck) [reply src/components/CardOverview.vue:30]: @MartinPirate ping

**Comment by mwalbeck** [comment]
I've tried to give it a spin, but nothing is showing up for me, is there something I'm missing? @MetaProvide/adminly

**Comment by oliveiraigorm** [comment]
> I've tried to give it a spin, but nothing is showing up for me, is there something I'm missing? @MetaProvide/adminly

It was working fine before but I believe the component was removed by mistake on this commit https://github.com/MetaProvide/adminly_clients/pull/8/commits/64bd5c37a3ac7bd75ae19af66ed9a061dd356f69

@MartinPirate

**Comment by MartinPirate** [comment]
@mwalbeck  please test now

File: src/components/CardOverview.vue [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+			type: Float32Array,
+			default: 0,
+		},
+	},
+};
+</script>
+
+<style lang="css" scoped>
+.card {
+	box-shadow: 0px 0px 10.6566px rgba(0, 0, 0, 0.15);
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:89]: Whenever something is 0, just use 0 with no units. Also, for simplicity lets try and keep the numbers round, so use 11 for the box-shadow.

File: src/components/CardOverview.vue [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+	border-radius: 17.0505px;
+	background: #ffffff;
+	width: 200px;
+	height: 177.96px;
+	flex: 1;
+}
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:95]: If you add a `margin-right: 2rem` then we get some nice spacing between the boxes and still rely on flexbox for the actual sizing, so the elements are more dynamic.

File: src/components/CardOverview.vue [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:96]: Lastly we should add a new rule
```
.card:last-of-type {
	margin-right: 0;
}
```
That way we get the intended look.

File: src/components/CardOverview.vue [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+.card-content {
+	padding: 18px;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:98]: It's generally preferable to use relative units for padding, so here `1rem` will do.

File: src/components/CardOverview.vue [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+.card-header {
+	width: 60%;
+	height: 20px;
+	font-style: normal;
+	font-weight: 400;
+	font-size: 12.7879px;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:106]: For font-size we also want to use relative units instead, so here we can use `0.8rem`

File: src/components/CardOverview.vue [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+.card-data {
+	margin-top: 7px;
+}
+
+.card-data p {
+	height: 41px;
+	font-style: normal;
+	font-weight: 700;
+	font-size: 31.9697px;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:119]: Use `2rem` instead.

File: src/components/CardOverview.vue [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+	line-height: 40px;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:120]: Use `2.5rem` instead

File: src/components/CardOverview.vue [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+	color: #8e8e8e;
+}
+
+.card-data span svg {
+	margin-left: 4px;
+}
+
+hr {
+	border: 1.06566px solid #eeeeee;
+	height: 0px;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:142]: Unneeded

File: src/components/CardOverview.vue [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+	font-weight: 400;
+	font-size: 12.7879px;
+	line-height: 19px;
+	align-items: center;
+	text-align: center;
+	margin-left: 8px;
+	color: #8e8e8e;
+}
+
+.card-data span svg {
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:136]: Not really needed

File: src/components/CardOverview.vue [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+.card-data p {
+	height: 41px;
+	font-style: normal;
+	font-weight: 700;
+	font-size: 31.9697px;
+	line-height: 40px;
+	color: #1f1f1f;
+}
+
+.card-data span {
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:124]: The only rules needed here are:
```
font-weight: 400;
font-size: 0.8rem;
color: #8e8e8e;
```

File: src/components/CardOverview.vue [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+								<path
+									d="M0.988891 4.93102L5.22483 0.714441C5.4429 0.497367 5.79646 0.497367 6.01453 0.714441L10.2505 4.93102C10.6022 5.28119 10.3531 5.87992 9.85561 5.87992H1.38374C0.886258 5.87992 0.637116 5.28119 0.988891 4.93102Z"
+									fill="#47B881"
+								/>
+							</svg>
+						</span>
+					</p>
+				</div>
+				<hr />
+				<div class="card-history">
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:29]: I would replace this card-history section with:
```
<div class="card-history">
	<div class="card-history-entry history-yesterday">
		<span>Yesterday</span>
		<span>
			{{ yesterdayCount }} ({{
				yesterdayPercentageChange
			}}
			%)
		</span>
	</div>
	<div class="card-history-entry history-last-week">
		<span>Last week</span>
		<span>
			{{ lastWeekCount }} (
			{{ lastWeekPercentageChange }} %)
		</span>
	</div>
</div>
```
That accompanied by the CSS suggestions make the setup simpler and easier to maintain.

File: src/components/CardOverview.vue [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+	margin-left: 4px;
+}
+
+hr {
+	border: 1.06566px solid #eeeeee;
+	height: 0px;
+	margin-top: 12px;
+}
+
+.card-history {
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:146]: Replace this rule and the rest related to the card-history with the following:

```
.card-history {
	display: flex;
	flex-direction: column;
	font-size: 0.8rem;
}

.card-history .history {
	display: flex;
	justify-content: space-between;
}
```

**Review by mwalbeck — CHANGES_REQUESTED** [review]

File: src/components/CardOverview.vue [hunk 18]
```diff
@@ -86,93 +83,53 @@ export default {
...
 
 .card-content {
-	padding: 18px;
+	padding: 1.1rem;
 }
 
 .card-header {
-	width: 60%;
-	height: 20px;
+	height: 1.1rem;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:97]: Is there a specific reason why you kept the height parameter?

File: src/components/CardOverview.vue [hunk 19]
```diff
@@ -86,93 +83,53 @@ export default {
...
 	font-style: normal;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:98]: This one can also be removed

File: src/components/CardOverview.vue [hunk 20]
```diff
@@ -86,93 +83,53 @@ export default {
...
 	font-weight: 400;
```
Comment (mwalbeck) [inline src/components/CardOverview.vue:99]: As well as this one

File: src/components/CardOverview.vue (hunk 19 above)
Comment (mwalbeck) [reply src/components/CardOverview.vue:96]: ping @MartinPirate

File: src/components/CardOverview.vue (hunk 20 above)
Comment (mwalbeck) [reply src/components/CardOverview.vue:97]: ping @MartinPirate

**Review by mwalbeck — DISMISSED** [review]

**Review by mwalbeck — APPROVED** [review]

**Pull request merged**